airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -21,10 +21,17 @@
 
 
 class AmplitudeStream(HttpStream, ABC):
-
-    url_base = "https://amplitude.com/api/"
+    url_base = ""
     api_version = 2
 
+    def __init__(self, data_region: str, **kwargs):
+        super().__init__(**kwargs)
+        self.url_base = (
+            "https://analytics.eu.amplitude.com/api/"
+            if data_region == "EU Residency Server"
+            else "https://amplitude.com/api/"
+        )
```

Comment:
Would be better transform the `url_base` into a property function @mariamthiam. Check how Zendeks Sunshine does it: https://github.com/airbytehq/airbyte/blob/f11507b23a67f591a3c37abd8169d68c9aa1f1df/airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py#L26-L28

Reply:
Do you mind adding a link to Amplitude docs explaining about regions?

Reply:
Yes here is [API documentation](https://www.docs.developers.amplitude.com/analytics/apis/chart-annotations-api/#authorization) of Amplitude  , in the `Endpoints` section you have the different server.